syntax = "proto3";

package sekai.db;

import "sekai/db/proto/descriptor.proto";

message CharacterMissionV2AreaItem {
  option (master_db_file) = "characterMissionV2AreaItems.json";
  enum CharacterMissionType {
    UNKNOWN_CHARACTER_MISSION_TYPE   = 0;
    AREA_ITEM_LEVEL_UP_CHARACTER     = 1 [(json_value) = "area_item_level_up_character"];
    AREA_ITEM_LEVEL_UP_REALITY_WORLD = 2 [(json_value) = "area_item_level_up_reality_world"];
    AREA_ITEM_LEVEL_UP_UNIT          = 3 [(json_value) = "area_item_level_up_unit"];
  }

  enum Unit {
    UNKNOWN_UNIT   = 0;
    IDOL           = 1 [(json_value) = "idol"];
    LIGHT_SOUND    = 2 [(json_value) = "light_sound"];
    PIAPRO         = 3 [(json_value) = "piapro"];
    SCHOOL_REFUSAL = 4 [(json_value) = "school_refusal"];
    STREET         = 5 [(json_value) = "street"];
    THEME_PARK     = 6 [(json_value) = "theme_park"];
  }

  optional int32 id                                    = 1 [(json_name) = "id"];
  optional CharacterMissionType character_mission_type = 2 [(json_name) = "characterMissionType"];
  optional int32 area_item_id                          = 3 [(json_name) = "areaItemId"];
  optional int32 character_id                          = 4 [(json_name) = "characterId"];
  optional Unit unit                                   = 5 [(json_name) = "unit"];
}
